com.unity.sg2/Editor/GraphUI/DataModel/Blackboard/GraphDataVariableDeclarationModel.cs

```diff
@@ -112,6 +104,45 @@ public override void CreateInitializationValue()
                 {
                     InitializationModel.ObjectValue = Matrix4x4.identity;
                 }
+
+                if (DataType == ShaderGraphExampleTypes.Color)
+                {
+                    ContextEntry.AddField(ContextEntryEnumTags.kIsColor, true);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the settings applicable to this variable declaration.
+        /// </summary>
+        internal IEnumerable<VariableSetting> GetSettings()
+        {
+            // TODO (Joe): Enable slider mode when Range(min, max) display type can be generated.
+            // if (DataType == TypeHandle.Float)
+            // {
+            //     yield return VariableSettings.floatMode;
+            //
+            //     if (VariableSettings.floatMode.GetTyped(this) is ContextEntryEnumTags.FloatDisplayType.Slider)
+            //     {
+            //         yield return VariableSettings.rangeMin;
+            //         yield return VariableSettings.rangeMax;
+            //     }
+            // }
+
+            if (DataType == ShaderGraphExampleTypes.Color)
+            {
+                yield return VariableSettings.colorMode;
+            }
+
+            if (DataType == ShaderGraphExampleTypes.SamplerStateTypeHandle)
+            {
+                yield return VariableSettings.samplerStateFilter;
+                yield return VariableSettings.samplerStateWrap;
+            }
+
+            if (IsExposable)
+            {
+                yield return VariableSettings.shaderDeclaration;
```

Comment:
Just a note that this sort of pattern, where we switch on what to do based on a type, becomes troublesome for every new type we add- and esp. if we allow a user to add new types.

Just another indicator that our Type System is missing some things.